modules/javafx.controls/src/test/java/test/javafx/scene/control/skin/LabelSkinTest.java

```diff
@@ -2117,6 +2153,24 @@ public void underscoreAtEndOfTextIsNotAMnemonic() {
         assertEquals("foo_", LabelSkinBaseShim.getText(label).getText());
     }
 
+    @Test
+    public void mnemonicParsingStopsAfterFirstSimpleMnemonic() {
+        label.setMnemonicParsing(true);
+        label.setText("_foo _bar _qux");
+        label.autosize();
+        skin.updateDisplayedText();
+        assertEquals("foo _bar _qux", LabelSkinBaseShim.getText(label).getText());
```

Comment:
Can you verify here which letter / index is picked as mnemonic? (Also in the other tests).

Reply:
I don't see an easy way to do that, and I'm not in favor of making private implementation details package-public just to test some internal state. Of course, mnemonic support should have been designed in a way that is more easily testable, but this PR is not the place to do that.

Reply:
could be tested indirectly, though not in isolation: 

- access the actual mnenomic via accessibleAttribute
- test whether labelFor/action is working as expected when firing an alt-mnemonic onto the scene

Reply:
How about adding a JUnit test for TextBinding instead?  The class is sufficiently complicated to warrant one, and it has quite a few branches to cover.  I think testing it through a `Control` is a bit too high level.

Something like:

    TextBinding tb = new TextBinding("complicated_mnemonic__example_(s)__");

    assertEquals("m", tb.getMnemonic());
    assertEquals(KeyCombination.M, tb.getMnemonicKeyCombination());
    assertEquals(12, tb.getMnemonicIndex());

Reply:
Just noticed @kleopatra that you suggested the same thing, I definitely agree that the place to test this is in a test for `TextBinding`.